frameworks/keyed/fntags/src/Main.js

```diff
@@ -29,50 +29,58 @@ const Button = ( id, title, onclick ) =>
        h( 'button', { id, type: 'button', class: 'btn btn-primary btn-block', onclick: onclick }, title )
     )
 
-let rowTemplate =
-    h( 'tr',
-       h( 'td', { class: 'col-md-1' } ),
-       h( 'td', { class: 'col-md-4' }, h( 'a' ) ),
-       h( 'td', { class: 'col-md-1' },
-          h( 'a',
-             h( 'span', { class: 'glyphicon glyphicon-remove', 'aria-hidden': 'true' } )
-          )
-       ),
-       h( 'td', { class: 'col-md-6' } )
-    )
-let selected = null
-
 const row = ( item ) => {
-    const tr = rowTemplate.cloneNode( true )
-    const id = tr.firstChild
-    const label = id.nextSibling.firstChild
-    const removeBtn = id.nextSibling.nextSibling
-
-    removeBtn.addEventListener( 'click', (e) => {
-        e.preventDefault()
-        e.stopPropagation()
-        tr.replaceWith('')
-        item.reset()
-        item(null)
+    const id = h( 'td', {
+                      class: 'col-md-1'
+                  },
+                  item().id
+    )
+    let currentLabel = item().label
+    let label = h( 'a', item().label )
+    let removeBtn = h( 'span', {
+        onclick: ( e ) => {
+            e.preventDefault()
+            e.stopPropagation()
+            data( data().filter( d => d().id !== item().id ) )
+        },
+        class: 'glyphicon glyphicon-remove', 'aria-hidden': 'true'
     } )
 
-    tr.addEventListener( 'click', () => {
-        if( selected ) selected.patch( { selected: false } )
-        tr.className = 'danger'
-        item.patch( { selected: true } )
-        selected = item
-    } )
+    let isSelected = item().selected
+    const tr = h( 'tr', {
+                      id: item().id,
+                      class: isSelected ? 'danger' : '',
+                      onclick: () => {
+                          let currentSelection = data().find( d => d().selected )
+                          if( currentSelection ) currentSelection.patch( { selected: false } )
+                          item.patch( { selected: true } )
+                      }
+                  },
+                  id,
+                  h( 'td', { class: 'col-md-4' }, label ),
+                  h( 'td', { class: 'col-md-1' },
+                     h( 'a',
+                        removeBtn
+                     )
+                  ),
+                  h( 'td', { class: 'col-md-6' } )
+    )
 
-    tr.setAttribute( 'id', item().id.toString() )
+    item.subscribe( () => {
+        if( isSelected && !item().selected ) {
+            tr.className = ''
+        } else if( !isSelected && item().selected ) {
+            tr.className = 'danger'
+        }
+        isSelected = item().selected
 
-    const update = () => {
-        tr.className = item().selected ? 'danger' : ''
-        label.innerText = item().label
-        id.innerText = item().id
-    }
+        if( currentLabel != item().label ) {
+            label.innerText = item().label
+            currentLabel = item().label
+        }
+    } )
```

Comment:
The implementation definitely looks cleaner. But this is the part that is going to draw heat. It is data-driven in the sense that you are subscribing to the model. But I guess what I'd be wondering is does the HyperScript not have a way to subscribe to these changes without escaping into this imperative chunk of direct DOM manipulation?

I looked at the library to see if I could find a way of representing what I mean but I think the answer is no.  Like a few other libraries this falls into a place where it is intended use. It's a tricky one since in one sense it is what any reactive library does under the hood and could do in the open. Writing it this way leaves it easy to optimize for the specific benchmark. It isn't unlike what Solid and Surplus were doing previously, but it's also exactly what categorized them under 772.

Reply:
No, it doesn't. That's essentially the point of this library, provide reactive updates in an obvious way. The current voodoo magic methods are so non-obvious that it makes the learning curve more like a learning cliff for a lot of these frameworks.

IMO, it looks like a duck, it quacks like a duck, it's a duck. 

All I really wanna know is if it's faster or not.

Reply:
I'd also like to do a marketing flip on this statement "leaves it easy to optimize for the specific benchmark" and change it to more accurately (and less pessimistically) say "leaves it easy to optimize for any specific use case".